### ISSUE TITLE
IOTEDGE_DEVICEID and IOTEDGE_MODULEID environment variables are not set

### ISSUE THREAD

**Issue by AMCN41R** [Azure/iotedgehubdev#175]
I am trying to access the device id from a C# module using `Environment.GetEnvironmentVariable("IOTEDGE_DEVICEID")` as descibed in the following issues...

[iotedge-issues-633](https://github.com/Azure/iotedge/issues/633) 
[azure-iot-sdk-csharp-issues-515](https://github.com/Azure/azure-iot-sdk-csharp/issues/515)

However, these variables are never set when using the `iotedgehubdev` tool.

I am using `iotedgehubdev, version 0.7.0`

**Comment by adashen** [comment]
@AMCN41R Thanks for reporting the issue. We are discussing with SDK team to decide whether there is an API for user to use. Since currently the ENV is just a workaround. We will take action after we figure out the official way. We will keep you updated when there is a decision.

**Comment by kiranpradeep** [comment]
@adashen Is there an update on this? We are trying to invoke direct methods on other modules and that is not possible without `IOTEDGE_DEVICEID` being populated.

**Comment by adashen** [comment]
@kiranpradeep can you give more detail of the scenario? Currently there is no update from SDK side. And we would like to know more about your scenario. Besides, the quick workaround, is set the ENV in your execution environment to unblock you first.

**Comment by kiranpradeep** [comment]
@adashen Thanks for guiding. I will now proceed with setting ENV in deployment.debug.template.json.

The scenario was to invoke a direct method from moduleA to mobileB using [InvokeMethodAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.moduleclient.invokemethodasync?view=azure-dotnet). But that needed the device id from `IOTEDGE_DEVICEID` env.

**Comment by SLdragon** [comment]
Hi, @kiranpradeep , since there is no update from SDK side, and there is no official way to get IOTEDGE_DEVICEID env, so we will keep this issue open.
Just as adashen methioned, you can set env in deployment.debug.template.json for solution mode, and for single module mode, you can pass an env parameter through command line:

```
iotedgehubdev start -i "<module-inputs>" -e "<environment-variable>"
```

**Comment by Fredborg** [comment]
I tried a simple test. I created a new empty module called EnvVariableModule. In deployment.debug.template.json i confgiured the module with 1 Environment Variable called ENV_VAR like so:

`"EnvVariableModule": { "version": "1.0.0", "type": "docker", "status": "running", "restartPolicy": "always", "settings": { "image": "${MODULEDIR<../EnvVariableModule>.debug}", "createOptions": { "Env": [ "ENV_VAR=MyCoolVariable" ] } } } }
`
In program.cs I added the following line of code to retrieve the value of ENV_VAR like so:

`string environmentVar = Environment.GetEnvironmentVariable("ENV_VAR");`

When starting up environmentVar = null

Why is this not working?

**Comment by Noppey** [comment]
When setting an environment variable, how does one interpolate the device id as the proper value? Let's say in solution mode.

**Comment by konichi3** [comment]
Hi 

Thank you for filing the issue.

We are looking to clean up the reported issues that are obsolete such that our team can focus on the latest issues reported.
As such, we are going to close the issue for now.

If the issue is still blocking or critical to you, please file a new issue and we will do our best to respond as soon as possible.
When you file a new issue, please ensure following additional information are provided.

o	If you believe it’s a bug, provide detail repro steps. 
o	If it’s a feature request, please note as such
o	Expected outcome
o	Is this a blocking issue?

Thank you,
Azure IoT Developer and Device team
